sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -121,7 +120,7 @@ case class BroadcastExchangeExec(
       // and only 70% of the slots can be used before growing in UnsafeHashedRelation,
       // here the limitation should not be over 341 million.
       (BytesToBytesMap.MAX_CAPACITY / 1.5).toLong
-    case _ => 512000000
```

Comment:
This hardcoded value is probably the physical limitation. If we want fine-grained control over broadcast join decisions, we can probably add a config that is similar to `spark.sql.autoBroadcastJoinThreshold` and is based on number of rows.

Reply:
It's a good idea. Maybe I can try to add the config. I'll revert the changes. Thanks so much!